pkg/cleaner/cleaner_test.go

```diff
@@ -0,0 +1,44 @@
+package cleaner
+
+import (
+	"testing"
+
+	"time"
+
+	"github.com/deis/builder/pkg/sys"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type namespace struct {
+}
+
+// Watch returns a watch.Interface that watches the requested namespaces.
+
+func (r *namespace) Watch(label labels.Selector, field fields.Selector, resourceversion string) (watch.Interface, error) {
+	nst := watch.NewFake()
+	go func() {
+		nst.Add(&api.Namespace{ObjectMeta: api.ObjectMeta{Name: "dir1"}})
+		nst.Modify(&api.Namespace{ObjectMeta: api.ObjectMeta{Name: "dir1"}})
+		nst.Delete(&api.Namespace{ObjectMeta: api.ObjectMeta{Name: "dir1"}})
+		nst.Stop()
+	}()
+	return nst, nil
+}
+
+func (r *namespace) IsAnAPIObject() {}
+
+func TestCleanerRun(t *testing.T) {
+	ns := &namespace{}
+	fs := sys.NewFakeFS()
+	dirhome := "/home/git"
+	fs.Files["/home/git/dir1.git"] = []byte{}
+	go Run(dirhome, ns, fs)
+	time.Sleep(5 * time.Second)
+	_, ok := fs.Files["/home/git/dir1.git"]
+	if ok {
+		t.Fatal("direcotry not deleted unit test failed")
+	}
+}
```

Comment:
@smothiki can you add a few other tests for failure cases. For example:

1. The watcher sends other objects, like pods or replication controllers. In those cases, the cleaner should do nothing
2. The watcher sends namespaces that don't exist in the fake filesystem
3. The watcher doesn't send any `DELETE` events